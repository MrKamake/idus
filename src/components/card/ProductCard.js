import React from 'react';
import PropTypes from 'prop-types';
import './ProductCard.scss';
import Rating from 'react-rating';
import { TiStarOutline, TiStarFullOutline } from 'react-icons/ti';

const ProductCard = ({ title, created_by, cover_image, rating, comment }) => {
  return (
    <div className="ui_card">
      <div className="ui_card__imgcover">
        <img className="ui_card__img" src={cover_image} alt="cover" />
      </div>
      <div className="ui_card__info">
        <span className="ui_card__label">{created_by}</span>
        <span className="ui_card__title">{title}</span>
        <div className="ui_card__foot">
          <span className="ui_card__hilight">Hilight</span>
          <span className="ui_card__cross-out">Cross out</span>
        </div>
      </div>
      {rating && (
        <div className="ui_card__rating">
          <div className="ui_rating_wrapper">
            <Rating
              className="ui_rating"
              readonly
              initialRating={rating}
              emptySymbol={<TiStarOutline size="20px" />}
              fullSymbol={<TiStarFullOutline size="20px" />}
            />
            {comment && <span className="ui_card__comment">{comment}</span>}
          </div>
        </div>
      )}
    </div>
  );
};

ProductCard.defaultProps = {
  title: 'untitle',
  created_by: 'unknown',
  cover_image: '',
  rating: 0,
  comment: ''
};

ProductCard.propTypes = {
  title: PropTypes.string.isRequired,
  created_by: PropTypes.string.isRequired,
  cover_image: PropTypes.string.isRequired,
  rating: PropTypes.number,
  comment: PropTypes.string
};

export default ProductCard;
