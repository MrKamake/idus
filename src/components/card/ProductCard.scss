.ui_card {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-left: 16px;
  margin-top: 16px;
  overflow: hidden;
  border-radius: 4px;
  border: 1px solid #d9d9d9;
  font-size: 12px;
  background: #f8f9fb;
  box-shadow: 0 1px 3px 0 hsla(0, 0%, 86.3%, 0.3);
  .ui_card__imgcover {
    overflow: hidden;
    .ui_card__img {
      width: 196.391px;
    }
  }
  .ui_card__info {
    padding: 8px 10px;
    .ui_card__label {
      display: block;
      margin-bottom: 4px;
      width: 100%;
      height: 15px;
      font-size: 12px;
      color: #999;
      white-space: nowrap;
    }
    .ui_card__title {
      display: block;
      display: -webkit-box;
      height: 42px;
      font-size: 14px;
      font-weight: 400;
      line-height: 1.5;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
      color: #333;
    }
    .ui_card__foot {
      .ui_card__hilight {
        font-size: 12px;
        color: #dc5850;
      }
      .ui_card__cross-out {
        margin-left: 3px;
        font-size: 10px;
        text-decoration-line: line-through;
        color: #acacac;
      }
    }
  }
  .ui_card__rating {
    display: table;
    padding: 0 10px;
    width: 100%;
    height: 56px;
    border-top: 1px solid #d9d9d9;
    box-sizing: border-box;
    vertical-align: middle;
    .ui_rating_wrapper {
      display: table-cell;
      vertical-align: middle;
      width: 176px;
      margin-bottom: 0;
      .ui_rating {
        padding: 6px 0 4px;
        width: 100%;
        color: #ffc801;
      }
      .ui_card__comment {
        color: #666;
        display: -webkit-box;
        height: 18px;
        font-size: 12px;
        line-height: 1.5;
        -webkit-line-clamp: 1;
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }
  }
}
