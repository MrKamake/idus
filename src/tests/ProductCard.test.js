import React from 'react';
import { shallow } from 'enzyme';
import ProductCard from '../components/card/ProductCard';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import 'react-lazy-load-image-component/src/effects/blur.css';

describe('ProductCard', () => {
  describe('<ProductCard />', () => {
    let wrapper, props;

    beforeEach(() => {
      props = {
        title: 'Card Title',
        created_by: 'Card Label',
        cover_image: ''
      };
      wrapper = shallow(<ProductCard {...props} />);
    });

    it('matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should contain elements (1)', () => {
      expect(
        wrapper.contains(
          <LazyLoadImage
            effect="blur"
            className="ui_card__img"
            src=""
            alt="cover"
          />
        )
      ).toBe(true);
      expect(
        wrapper.contains(<span className="ui_card__label">Card Label</span>)
      ).toBe(true);
      expect(
        wrapper.contains(<span className="ui_card__title">Card Title</span>)
      ).toBe(true);
      expect(wrapper.find('.ui_rating').length).toBe(0);
      expect(wrapper.find('.ui_card__comment').length).toBe(0);
    });

    it('should contain elements (2)', () => {
      props = {
        title: 'Title',
        created_by: 'Label',
        cover_image: 'IMG',
        rating: 3
      };
      wrapper = shallow(<ProductCard {...props} />);
      expect(
        wrapper.contains(
          <LazyLoadImage
            effect="blur"
            className="ui_card__img"
            src="IMG"
            alt="cover"
          />
        )
      ).toBe(true);
      expect(
        wrapper.contains(<span className="ui_card__label">Label</span>)
      ).toBe(true);
      expect(
        wrapper.contains(<span className="ui_card__title">Title</span>)
      ).toBe(true);
      expect(wrapper.find('.ui_rating').length).toBe(1);
      expect(wrapper.find('.ui_card__comment').length).toBe(0);
    });

    it('should contain elements (3)', () => {
      props = {
        title: 'ProductCard Title',
        created_by: 'Seongjoon Kim',
        cover_image: 'mountain',
        rating: 3,
        comment:
          'Lorem ipsum dolor sit amet consectetur adipisicing elits graitudegupo'
      };
      wrapper = shallow(<ProductCard {...props} />);
      expect(
        wrapper.contains(
          <LazyLoadImage
            effect="blur"
            className="ui_card__img"
            src="mountain"
            alt="cover"
          />
        )
      ).toBe(true);
      expect(
        wrapper.contains(<span className="ui_card__label">Seongjoon Kim</span>)
      ).toBe(true);
      expect(
        wrapper.contains(
          <span className="ui_card__title">ProductCard Title</span>
        )
      ).toBe(true);
      expect(wrapper.find('.ui_rating').length).toBe(1);
      expect(wrapper.find('.ui_card__comment').length).toBe(1);
      expect(
        wrapper.contains(
          <span className="ui_card__comment">
            Lorem ipsum dolor sit amet consectetur adipisicing elits
            graitudegupo
          </span>
        )
      ).toBe(true);
    });
  });
});
