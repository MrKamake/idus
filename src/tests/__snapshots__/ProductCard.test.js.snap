// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard <ProductCard /> matches snapshot 1`] = `
<div
  className="ui_card"
>
  <div
    className="ui_card__imgcover"
  >
    <t
      afterLoad={[Function]}
      alt="cover"
      beforeLoad={[Function]}
      className="ui_card__img"
      delayMethod="throttle"
      delayTime={300}
      effect="blur"
      placeholderSrc={null}
      src=""
      threshold={100}
      useIntersectionObserver={true}
      visibleByDefault={false}
      wrapperClassName=""
    />
  </div>
  <div
    className="ui_card__info"
  >
    <span
      className="ui_card__label"
    >
      Card Label
    </span>
    <span
      className="ui_card__title"
    >
      Card Title
    </span>
    <div
      className="ui_card__foot"
    >
      <span
        className="ui_card__hilight"
      >
        Hilight
      </span>
      <span
        className="ui_card__cross-out"
      >
        Cross out
      </span>
    </div>
  </div>
  0
</div>
`;
